lib/can/include/CANMsg.h

```diff
@@ -1,14 +1,16 @@
 #pragma once
 
-#include <map>
-
+#include "LookupTable.h"
 #include "hw_bridge.h"
 #include "mbed.h"
 
 class CANMsg : public CANMessage {
+ private:
+  using CAN_Message::id;
  public:
-  typedef mbed_error_status_t (*CANMsgHandler)(CANMsg &);
-  typedef std::map<HWBRIDGE::CANID, CANMsg::CANMsgHandler> CANMsgHandlerMap;
+  using CANMsgHandler = mbed_error_status_t (*)(CANMsg &);
+  using CANMsgHandlerMap = LookupTable<HWBRIDGE::CANID, CANMsg::CANMsgHandler,
+                                       +[](CANMsg &)->mbed_error_status_t {return MBED_SUCCESS;}>;
```

Comment:
This shouldn't be a success

Reply:
MBED_OK?

Reply:
It should be some error cuz it got called with some ID that doesnt exist

Reply:
In fact, it probably should be a very serious error (fatal?) because it should be impossible to get to that point. This default only gets called when the can id provided doesnt have an associated function. This means either there isnt a function in the map for a valid id or a lookup was performed for an invalid Id (which should be impossible because we should be filtering for only valid ids we want). So both scenarios are just major programmer errors